pandas/tests/frame/methods/test_cov_corr.py

```diff
@@ -175,6 +175,15 @@ def test_corr_int(self):
         df3.cov()
         df3.corr()
 
+    @td.skip_if_no_scipy
+    @pytest.mark.parametrize("meth", ["pearson", "spearman", "kendall"])
+    def test_corr_axes(self, meth):
+        # https://github.com/pandas-dev/pandas/issues/35002
+        df = pd.DataFrame(np.random.normal(size=(10, 4)))
+        expected = df.T.corr(meth, axis=0)
```

Comment:
I think it's usually encouraged to explicitly write out the expected DataFrame so that `expected` doesn't go down the same code path as `result`

Reply:
Yeah I could do that but, wouldn't it just be a test for Dataframe.corr function itself. Since the original operations to be done on matrix itself are left unchanged.

Personally I don't think explicitly writing Dataframe in this case is needed, unless (as you suggested) instead of taking a transpose we implement a workaround involving changing the main function itself.

Reply:
OK sure, perhaps wait for others' comments then

Reply:
The test comes close to being circular but I think it's probably okay here. In this case it's hard to explicitly construct the expected DataFrame for all methods "from scratch" without either trivial input data or messy juggling of different scipy functions.